src/dune_rules/findlib/findlib.mli

```diff
@@ -51,10 +51,25 @@ module Config : sig
 
   val to_dyn : t -> Dyn.t
 
-  val load :
-    Path.Outside_build_dir.t -> toolchain:string -> context:string -> t Memo.t
+  (** Finds the library search paths for this configuration, prepending
+      [OCAMLPATH] if set *)
+  val path : t -> Path.t list
 
-  val get : t -> string -> string option
+  (** Finds program [prog] for this configuration, if it exists *)
+  val tool : t -> prog:string -> Path.t option Memo.t
 
-  val env : t -> Env.t
+  val ocamlpath_sep : char
+
+  (** Read and parse the [OCAMLPATH] environment variable *)
+  val ocamlpath : Env.t -> Path.t list option
+
+  val extra_env : t -> Env.t
+
+  (** Load the findlib configuration for this [findlib_toolchain], if any. *)
+  val discover_from_env :
+       env:Env.t
+    -> which:(string -> Path.t option Memo.t)
+    -> ocamlpath:Path.t list
+    -> findlib_toolchain:Dune_engine.Context_name.t option
```

Comment:
is this really a requirement that we currently have? that a tool chain name should be a valid context name?

Reply:
I'm not sure what our limitations are. I changed it to a string, which should allow more names.